tests/general/FilenameBearTest.py

```diff
@@ -118,3 +119,50 @@ def test_auto_file_naming_convention_warning(self):
         self.assertEqual(log.message,
                          'The file naming convention could not be guessed. '
                          'Using the default "snake" naming convention.')
+
+    def test_file_prefix(self):
+        self.section['filename_prefix'] = 'pre'
+        self.check_validity(
+            self.uut, [''], filename='filename.xyz', valid=False)
+        self.check_validity(
+            self.uut, [''], filename='prefilename.xyz', valid=True)
+        self.check_results(
+            self.uut,
+            [''],
+            [Result.from_values('FilenameBear',
+                                'Filename does not use '
+                                'the prefix {!r}.'.format('pre'),
+                                severity=RESULT_SEVERITY.NORMAL,
+                                file='filename.xyz')],
+            filename='filename.xyz')
+
+    def test_file_suffix(self):
+        self.section['filename_suffix'] = 'fix'
+        self.check_validity(
+            self.uut, [''], filename='filename.xyz', valid=False)
+        self.check_validity(
+            self.uut, [''], filename='filenamesuffix.xyz', valid=True)
+        self.check_results(
+            self.uut,
+            [''],
+            [Result.from_values('FilenameBear',
+                                'Filename does not use '
+                                'the suffix {!r}.'.format('fix'),
+                                severity=RESULT_SEVERITY.NORMAL,
+                                file='filename.xyz')],
+            filename='filename.xyz')
+
+    def test_file_prefix_suffix(self):
+        self.section['filename_prefix'] = 'pre'
+        self.section['filename_suffix'] = 'fix'
+        self.check_results(
+            self.uut,
+            [''],
+            [Result.from_values('FilenameBear',
+                                '\n'.join(['- Filename does not use '
+                                           'the prefix {!r}.'.format('pre'),
+                                           '- Filename does not use '
+                                           'the suffix {!r}.'.format('fix')]),
+                                severity=RESULT_SEVERITY.NORMAL,
+                                file='filename.xyz')],
+            filename='filename.xyz')
```

Comment:
Actually I was searching for one, and discovered no tests use `diffs`. I was wondering why your tests did already pass, because you definitely generate results with diffs. I found out that `Result` doesn't take diffs into account when comparing with each other. This is super bad and needs definitely a fix. Filed an issue here: https://github.com/coala/coala/issues/4400

Because of this, you don't have to provide a diff as it's meaningless now (via the `diffs` parameter of `Result.__init__`), but we file an issue once we are able to do so :+1: